engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/TestFlinkProcessCompiler.scala

```diff
@@ -1,39 +1,82 @@
 import pl.touk.nussknacker.engine.definition.DefinitionExtractor.ObjectWithMethodDef
+import pl.touk.nussknacker.engine.expression.ExpressionEvaluator
 import pl.touk.nussknacker.engine.flink.api.exception.FlinkEspExceptionConsumer
 import pl.touk.nussknacker.engine.flink.api.process.{FlinkIntermediateRawSource, FlinkSourceTestSupport}
 import pl.touk.nussknacker.engine.flink.util.source.{CollectionSource, EmptySource}
+import pl.touk.nussknacker.engine.graph.expression.Expression
 import pl.touk.nussknacker.engine.process.exception.FlinkExceptionHandler
+import pl.touk.nussknacker.engine.spel.SpelExpressionParser
 import pl.touk.nussknacker.engine.testmode.ResultsCollectingListener
+import pl.touk.nussknacker.engine.variables.GlobalVariablesPreparer
 
-class TestFlinkProcessCompiler(creator: ProcessConfigCreator,
-                               inputConfigDuringExecution: Config,
+class TestFlinkProcessCompiler(modelData: ModelData,
                                collectingListener: ResultsCollectingListener,
                                process: CanonicalProcess,
-                               scenarioTestData: ScenarioTestData,
-                               objectNaming: ObjectNaming)
-  extends StubbedFlinkProcessCompiler(process, creator, inputConfigDuringExecution, diskStateBackendSupport = false, objectNaming, ComponentUseCase.TestRuntime) {
+                               scenarioTestData: ScenarioTestData)
+  extends StubbedFlinkProcessCompiler(process, modelData.configCreator, modelData.processConfig, diskStateBackendSupport = false, modelData.objectNaming, ComponentUseCase.TestRuntime) {
 
-  
   override protected def adjustListeners(defaults: List[ProcessListener], processObjectDependencies: ProcessObjectDependencies): List[ProcessListener] = {
     collectingListener :: defaults
   }
 
+  private val expressionEvaluator: (Expression, Parameter, NodeId) => ValidatedNel[PartSubGraphCompilationError, AnyRef] = {
+    val validationContext = GlobalVariablesPreparer(modelData.processWithObjectsDefinition.expressionConfig).emptyValidationContext(process.metaData)
+    val evaluator = ExpressionEvaluator.unOptimizedEvaluator(modelData)
+    val dumbContext = Context("dumb", Map.empty, None)
+    val expressionCompiler = ExpressionCompiler.withoutOptimization(modelData).withExpressionParsers {
+      case spel: SpelExpressionParser => spel.typingDictLabels
+    }
+    (expression: Expression, parameter: Parameter, nodeId: NodeId) => {
+      expressionCompiler
+        .compile(expression, Some(parameter.name), validationContext, parameter.typ)(nodeId)
+        .map { typedExpression =>
+          val param = evaluatedparam.Parameter(typedExpression, parameter)
+          evaluator.evaluateParameter(param, dumbContext)(nodeId, process.metaData).value
+        }
+    }
+  }
+
+  private def collectSamples(originalSource: Any, sourceWithTestSupport: FlinkSourceTestSupport[Object@unchecked], nodeId: NodeId): List[Object] = {
```

Comment:
In my opinion this method could be put just before `prepareDataForTest`. In my opinion it's more clear to put public, general methods at the beginning of the file and later detailed, private methods. WDYT?

Reply:
:+1: You're right, I've moved that definition

Reply:
I'm moving flink implementation to separate PR so I'm closing this thread..